nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/logging/repository/StandardLogRepository.java

```diff
@@ -82,14 +82,32 @@ public void addLogMessage(final LogLevel level, final String format, final Objec
         addLogMessage(level, formattedMessage, t);
     }
 
-    private void replaceThrowablesWithMessage(Object[] params) {
+    private void replaceThrowablesWithMessage(final Object[] params) {
         for (int i = 0; i < params.length; i++) {
-            if(params[i] instanceof Throwable) {
-                params[i] = ((Throwable) params[i]).getLocalizedMessage();
+            if (params[i] instanceof Throwable) {
+                params[i] = getCauses((Throwable) params[i]);
             }
         }
     }
 
+    private static String getCauses(final Throwable throwable) {
+        final StringBuilder builder = new StringBuilder();
+        if (throwable == null || throwable.getLocalizedMessage() == null) {
+            return "";
+        }
+        builder.append(throwable.getLocalizedMessage());
+        if (throwable.getCause() != null) {
+            builder.append("\n");
+        }
+        final Throwable firstCause = throwable.getCause();
+        for (Throwable t = firstCause; t != null; t = t.getCause()) {
+            builder.append("┗━▶ causes: ");
```

Comment:
The code appears to have a special character prior to `causes:` that should be removed and replaced with a standard separator of some kind.